regression/last_dev.py

```diff
@@ -0,0 +1,152 @@
+                    i = temp.index('--epochs')
+                    j = temp.index('--model')
+                    k = temp.index('--dataset')
+
+                    if '--qat-policy' in temp:
+                        x = temp.index('--qat-policy')
+                        temp[x+1] = "policies/qat_policy.yaml"
+                    else:
+                        temp.insert(-1, ' --qat-policy policies/qat_policy.yaml')
+
+                    log_model = temp[j+1]
+                    log_data = temp[k+1]
+
+                    if log_model == "ai87imageneteffnetv2":
+                        num = temp.index("--batch-size")
+                        temp[num+1] = "128"
+
+                    log_name = temp[j+1] + '-' + temp[k+1]
+                    log_file_names.append(filename[:-3])
+
+                    if log_data == "FaceID":
+                        continue
+                    if log_data == "VGGFace2_FaceDetection":
+                        continue
+                    if log_data == "ai85tinierssdface":
+                        continue
+
+                    temp[i+1] = str(config[log_data][log_model]["epoch"])
+
+                    if '--deterministic' not in temp:
+                        temp.insert(-1, '--deterministic')
+
+                    temp.insert(-1, '--name ' + log_name)
+
+                    data_name = temp[k+1]
+                    if data_name in config and "datapath" in config[data_name]:
+                        path_data = config[log_data]["datapath"]
+                        temp.insert(-1, '--data ' + path_data)
+
+                    temp.append("\n")
+                    contents = joining(temp)
+                    output_file.write(contents)
+
+
+def dev_checkout():
+    """
+    Checkout the last developed code
+    """
+    repo_url = "https://github.com/MaximIntegratedAI/ai8x-training.git"
+    local_path = pathconfig['local_path']
+
+    try:
+        repo = git.Repo(local_path)
+    except InvalidGitRepositoryError:
+        repo = git.Repo.clone_from(repo_url, local_path, branch="develop", recursive=True)
+
+    commit_hash = repo.heads.develop.object.hexsha
+    commit_num_path = pathconfig['commit_num_path']
+
+    try:
+        with open(commit_num_path, "r", encoding='utf-8') as file:
+            saved_commit_hash = file.read().strip()
+    except FileNotFoundError:
+        saved_commit_hash = ""
+
+    if commit_hash != saved_commit_hash:
+        with open(commit_num_path, "w", encoding='utf-8') as file:
+            file.write(commit_hash)
+            repo.remotes.origin.pull("develop")
+
+            dev_scripts(script_path, output_file_path)
+            cmd_command = "bash " + output_file_path
+            subprocess.run(cmd_command, shell=True, check=True)
+
+            path_command = "cd " + local_path
+            subprocess.run(path_command, shell=True, check=True)
+
+            source_path = "/home/asyaturhal/actions-runner/_work/ai8x-training/ai8x-training/logs/"
```

Comment:
Should not use a hard coded personal home here

Reply:
It is fixed.